src/plots/plots.js

```diff
@@ -1315,6 +1315,19 @@ plots.supplyLayoutGlobalDefaults = function(layoutIn, layoutOut, formatObj) {
         return Lib.coerce(layoutIn, layoutOut, plots.layoutAttributes, attr, dflt);
     }
 
+    // Special treatment for title.y that can be 'auto'
+    // or a number between 0 and 1.
+    function coerceTitleY() {
```

Comment:
cc @alexcjohnson here's another attribute that would benefit from adding a special string value to `valType: 'number'` as discussed in https://github.com/plotly/plotly.js/pull/3109#discussion_r227543929 and https://github.com/plotly/plotly.js/issues/2834#issuecomment-433529733

Reply:
[no need to tackle this in this PR of course @rmoestl]

Reply:
That would be great, but would this not work with `type: 'number', dflt: 'auto'`? Anyway as is I don't see where this version brings in the bounds `[0, 1]`?

Reply:
Oh, my bad. I removed min and max from the attribute definition after I think plotschema_test was having failures. It comlained that min and max were defined for `valType: 'any'`. I should have stayed with `valType: 'number'` in the first place. I should never have removed min and max without at least checking the bounds in the custom `coerceTitleY`.

Reply:
> `type: 'number', dflt: 'auto'`

I quickly tested that and it looks good I believe. I'll add another commit.

Reply:
See commit https://github.com/plotly/plotly.js/pull/3276/commits/5fb7b8f4aa180ee894dce63646510407fb535777